src/compat/dbtype_function.c

```diff
@@ -32,6 +32,7 @@
 #include "memory_alloc.h"
 
 // hidden functions (suppress -Wmissing-prototypes)
-int db_make_db_char (DB_VALUE * value, const INTL_CODESET codeset, const int collation_id, char *str, const int size);
+int db_make_db_char (DB_VALUE * value, const INTL_CODESET codeset, const int collation_id, const char *str,
```

Comment:
str is not really `const char *`. see function definition:

```c
  value->data.ch.medium.buf = (char *) str;
```

It may be dangerous to assume that the argument is const.